scipy/optimize/lbfgsb_src/lbfgsb.f

```diff
@@ -2559,8 +2559,20 @@ subroutine lnsrlb(n, l, u, nbd, x, f, fold, gd, gdold, g, d, r, t,
          if (stp .eq. one) then
             call dcopy(n,z,1,x,1)
          else
+c        take step and prevent rounding error beyond bound GH10942
             do 41 i = 1, n
-               x(i) = stp*d(i) + t(i)
+               if (nbd(i) .ne. 0) then
```

Comment:
Looks ok, but it's possible to write it shorter:
```
               x(i) = stp*d(i) + t(i)
               if (nbd(i).eq.1.or.nbd(i).eq.2) x(i) = max(x(i), l(i))
               if (nbd(i).eq.2.or.nbd(i).eq.3) x(i) = min(x(i), u(i))
```
The `!` comments are Fortran 90+, whereas the rest of the file is F77, so probably best to avoid them.

Reply:
probably just "old habits die hard"